src/utils/crypto.js

```diff
@@ -39,24 +39,34 @@ function hmacJsonForPskCondition (obj, sharedSecret) {
 
 // turn object into encrypted buffer
 function aesEncryptObject (obj, sharedSecret) {
+  return aesEncryptBuffer(Buffer.from(JSON.stringify(obj), 'utf8'))
+}
+
+// turn buffer into encrypted buffer
+function aesEncryptBuffer (buffer, sharedSecret) {
   const pskEncryptionKey = hmac(sharedSecret, PSK_ENCRYPTION_STRING)
   const cipher = crypto.createCipher('aes-256-ctr', pskEncryptionKey)
 
   return Buffer.concat([
-    cipher.update(Buffer.from(JSON.stringify(obj), 'utf8')),
+    cipher.update(buffer),
```

Comment:
https://nodejs.org/api/crypto.html#crypto_class_cipher the documentation on the cipher class says the complete ciphertext is `cipher.update(buffer) + cipher.final()`